start.php

```diff
@@ -75,16 +75,7 @@ function registration_randomizer_tarpit($wait = 5) {
 	elgg_set_plugin_setting($setting_name, $count + 1, 'registration_randomizer');
 	registration_randomizer_log("Tarpitting $ip for $wait seconds after $count failures.", false);
 
-	if ($wait > 0) {
-		// close mysql connections for the time of a sleep
-		mysql_close(_elgg_services()->db->getLink('read'));
```

Comment:
According to https://github.com/brettp/elgg-registration-randomizer/pull/2, you have to close the connection or the server hangs.